<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Chart</class>
 <widget class="QWidget" name="Chart">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1118</width>
    <height>587</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0" rowspan="2" colspan="6">
    <widget class="QCustomPlot" name="chartWidget" native="true">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
    </widget>
   </item>
   <item row="0" column="6">
    <widget class="QFrame" name="frame">
     <property name="minimumSize">
      <size>
       <width>201</width>
       <height>221</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>201</width>
       <height>221</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_4">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>目标值:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QTextBrowser" name="target_val_Text">
          <property name="styleSheet">
           <string notr="true">QLineEdit {
	/* 边框1px宽，颜色为深紫色 */
    border: 1px solid rgb(41, 57, 85);
	/* 给定3px边框圆角 */
    border-radius: 3px;
	/* 背景色定为白色吧 */
    background: white;
	/* 文本的大小 */
    font-size: 14px ;
}
 
QLineEdit:hover {
	/* 鼠标悬停时，我们将编辑框的边框设置为蓝色 */
    border: 1px solid blue; 
}</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>实际值:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QTextBrowser" name="fact_val_Text"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_6">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>比例 P:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="pLineEdit"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_7">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>积分 I:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="iLineEdit"/>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_8">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>微分 D:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDoubleSpinBox" name="dLineEdit"/>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_11">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>允许误差:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QSpinBox" name="allowable_error_QSpinBox"/>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_9">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>电机死区:</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QSpinBox" name="dead_zone_spinBox"/>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_10">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>I分离阈值:</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QSpinBox" name="I_Separate_spinBox"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="6">
    <widget class="QFrame" name="frame_2">
     <property name="minimumSize">
      <size>
       <width>201</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>201</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0" colspan="2">
       <widget class="QDial" name="dial">
        <property name="wrapping">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="readPidButton">
        <property name="text">
         <string>读取PID</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="writePidButton">
        <property name="text">
         <string>写入PID</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="serialTipLabel">
     <property name="font">
      <font>
       <family>宋体</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 0, 0);</string>
     </property>
     <property name="text">
      <string>串口未开启</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="1">
      <widget class="QComboBox" name="baudComboBox">
       <property name="styleSheet">
        <string notr="true">QComboBox {
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 2px 1px 2px;
	max-width: 6em;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border-left-width: 1px;
    border-left-color: darkgray;
    border-left-style: solid;
    border-top-right-radius: 3px; 
    border-bottom-right-radius: 3px;
}

QComboBox::down-arrow {
    image: url(://image/down_arrow_16.png);
}
</string>
       </property>
       <item>
        <property name="text">
         <string>4800</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>9600</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>19200</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>38400</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>57600</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>115200</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>128000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>256000</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QComboBox" name="portComboBox">
       <property name="styleSheet">
        <string notr="true">QComboBox {
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 2px 1px 2px;
	max-width: 6em;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border-left-width: 1px;
    border-left-color: darkgray;
    border-left-style: solid;
    border-top-right-radius: 3px; 
    border-bottom-right-radius: 3px;
}

QComboBox::down-arrow {
    image: url(://image/down_arrow_16.png);
}
</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="portLabel">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>9</pointsize>
        </font>
       </property>
       <property name="text">
        <string>选择串口：</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="baudLabel">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>9</pointsize>
        </font>
       </property>
       <property name="text">
        <string>选择波特率：</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="2">
    <widget class="QPushButton" name="openSerialButton">
     <property name="styleSheet">
      <string notr="true">/*按钮普通态*/
QPushButton
{
    /*背景颜色*/  
    background-color:rgb(255 , 255 , 255);
    /*边框圆角半径为8像素*/ 
    border-radius:8px;
}

/*按钮停留态*/
QPushButton:hover
{
    /*背景颜色*/  
    background-color:rgb(197, 212, 244);
}

/*按钮按下态*/
QPushButton:pressed
{
    /*背景颜色*/  
    background-color: rgb(162, 188, 228);
    /*左内边距为3像素，让按下时字向右移动3像素*/  
    padding-left:3px;
    /*上内边距为3像素，让按下时字向下移动3像素*/  
    padding-top:3px;
}</string>
     </property>
     <property name="text">
      <string>打开串口</string>
     </property>
    </widget>
   </item>
   <item row="2" column="3">
    <widget class="QPushButton" name="pauseUpdateButton">
     <property name="styleSheet">
      <string notr="true">/*按钮普通态*/
QPushButton
{
    /*背景颜色*/  
    background-color:rgb(255 , 255 , 255);
    /*边框圆角半径为8像素*/ 
    border-radius:8px;
}</string>
     </property>
     <property name="text">
      <string>暂停更新</string>
     </property>
    </widget>
   </item>
   <item row="2" column="4">
    <layout class="QGridLayout" name="gridLayout_7">
     <item row="0" column="0">
      <widget class="QLabel" name="xLabel">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>10</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>X: 0.00</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="yLabel">
       <property name="font">
        <font>
         <family>宋体</family>
         <pointsize>10</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Y: 0.00</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="5">
    <widget class="QPushButton" name="windowTopButton">
     <property name="styleSheet">
      <string notr="true">/*按钮普通态*/
QPushButton
{
    /*背景颜色*/  
    background-color:rgb(255 , 255 , 255);
    /*边框圆角半径为8像素*/ 
    border-radius:8px;
}</string>
     </property>
     <property name="text">
      <string>窗口置顶</string>
     </property>
     <property name="shortcut">
      <string/>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
